test/extensions/filters/http/alternate_protocols_cache/filter_integration_test.cc

```diff
@@ -52,22 +55,46 @@ name: alternate_protocols_cache
     config_helper_.prependFilter(filter);
 
     upstream_tls_ = true;
+    // This configures the upstream to use the connection grid (automatically
+    // selecting protocol and allowing HTTP/3)
     config_helper_.configureUpstreamTls(/*use_alpn=*/true, /*http3=*/true, alt_cache);
 
     HttpProtocolIntegrationTest::initialize();
   }
 
   void createUpstreams() override {
-    // The integration test infrastructure does not know how to create mixed protocol upstreams.
-    // It will have created an HTTP/3 upstream, so the test needs to create an additional
-    // HTTP/2 upstream manually.
-    ASSERT_EQ(upstreamProtocol(), Http::CodecType::HTTP3);
-    ASSERT_EQ(fake_upstreams_count_, 1);
+    // This function will create 2 upstreams, but Envoy will only point at the
+    // first, the HTTP/2 upstream.
```

Comment:
Is this commenting the behavior of createUptreams() or of skipPortUsageValidation()? If the former, how about moving to above the method name. If the latter.... wow skipPortUsageValidation() is a spectacularly confusing name for that :)

Reply:
moved comment, and added a new one :-)